### PR TITLE
[ARCTIC-1548] Replace log4j 1.2 to reload4j

### LINKED ISSUE #1548: [Bug]: Log4j 1.2 is a SocketServer class that is vulnerable to deserialization of untrusted data
### What happened?

[CVE-2019-17571](https://www.cvedetails.com/cve/CVE-2019-17571/)
Included in Log4j 1.2 is a SocketServer class that is vulnerable to deserialization of untrusted data which can be exploited to remotely execute arbitrary code when combined with a deserialization gadget when listening to untrusted network traffic for log data. This affects Log4j versions up to 1.2 up to 1.2.17.

### Affects Versions

master/0.4.x/0.5.x

### What engines are you seeing the problem on?

AMS, Flink, Spark

### How to reproduce

_No response_

### Relevant log output

_No response_

### Anything else

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by XBaith**
## Why are the changes needed?

Fix #1548

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -299,6 +304,12 @@
                 <version>${mockito.version}</version>
                 <scope>test</scope>
             </dependency>
+
+            <dependency>
+                <groupId>ch.qos.reload4j</groupId>
+                <artifactId>reload4j</artifactId>
+                <version>${reload4j.version}</version>
```
Comment (shidayang) [inline pom.xml:311]: what is the difference between ch.qos.reload4j.reload4j and org.slf4j.slf4j-reload4j

**Comment by zhongqishang** [comment]
AFAIK, Flink connector logging framework depends on log4j2 , And the log dependency will not be packaged into the jar.
I have two confuse:
1. What is the reason for package into jar
2. What is the reason for not upgrade to log4j2

@shidayang @XBaith

**Comment by zhoujinsong** [comment]
I agree with that we should upgrade the log4j dependency to log4j2 and not package the log dependency into the connector's JAR file.

BTW, does this issue only affect branch 0.4.x, which means the branch master does not have this issue.?

**Comment by XBaith** [comment]
> BTW, does this issue only affect branch 0.4.x, which means the branch master does not have this issue?

I believe that the branch master is also affected.

**Comment by shidayang** [comment]
> 
+1 upgrade to log4j2

**Comment by XBaith** [comment]
> +1 upgrade to log4j2, Could you upgrade log4j to log4j2?
OK, I will upgrade log4j

**Review by baiyangtx — CHANGES_REQUESTED** [review]
Did you forget to modify the Spark 3.2/3.3 module?

**Pull request closed without merging**